packages/dev/core/src/Engines/renderTargetWrapper.ts

```diff
@@ -198,11 +198,16 @@ export class RenderTargetWrapper {
         if (!this._textures) {
             this._textures = [];
         }
+        if (this._textures[index] === texture) {
+            return;
+        }
+
         if (this._textures[index] && disposePrevious) {
             this._textures[index].dispose();
         }
 
         this._textures[index] = texture;
+        texture.incrementReferences();

```

Comment:
I agree it can generate leaks, but it seemed to me that `setTexture`'s behavior of not incrementing the ref count was wrong, because it effectively does increment the ref count.
We never shared textures between framebuffers before, so the bug was silent, but now that we do isn't it the correct behavior ?

Reply:
The problem is that the texture we're passing to `setTexture` may be one we're sharing between framebuffers, in which case we need to increment the refcount, or a new texture we've just created, in which case we don't want to increment the refcount because it's already at 1. `setTexture` can't distinguish between the two cases.

Why not add an optional `incrementRefCount` parameter and set it to `false` to be backward compatible?